### PR TITLE
TABU: Update entries with same key but different component values

### LINKED ISSUE #6379: Table data (TABU): Update table data with same key but different component values
Currently abapGit is not able to update existing db table entries (TABU) with the same key values but different component values. Also see comment in class _zcl_abapgit_data_deserializer_, method _preview_database_changes_.
image

Please also note that the existing READ TABLE only compares the key fields not all components.

I provided a pull request.

### PR DESCRIPTION
**Pull request by mraht**
Closes #6379

### CODE REVIEW

File: src/data/zcl_abapgit_data_deserializer.clas.abap [hunk 1]
```diff
@@ -120,8 +121,10 @@ CLASS zcl_abapgit_data_deserializer IMPLEMENTATION.
     FIELD-SYMBOLS <lg_new> TYPE ANY TABLE.
     FIELD-SYMBOLS <ls_del> TYPE any.
     FIELD-SYMBOLS <ls_ins> TYPE any.
+    FIELD-SYMBOLS <ls_upd> TYPE any.
```
Comment (larshp) [inline src/data/zcl_abapgit_data_deserializer.clas.abap:124]: remove the todo comment in line 116?

**Comment by larshp** [comment]
is it possible to add a unit test for this change?

**Comment by mraht** [comment]
> is it possible to add a unit test for this change?

Hi @larshp, do you prefer one test for each of insert, update and delete function or all in one?

**Comment by larshp** [comment]
one test for each

more smaller tests is better IMHO

**Comment by mraht** [comment]
> one test for each
> 
> more smaller tests is better IMHO

Ok, and would it be enough to only test method preview_database_changes?

Also why does abapLint says "Forbidden void type: cl_aunit_assert"? We would have to use it to be able to run the tests?

**Comment by larshp** [comment]
> CL_AUNIT_ASSERT is obsolete

https://help.sap.com/docs/SAP_NETWEAVER_740/c238d694b825421f940829321ffa326a/4ec2efe26e391014adc9fffe4e204223.html?version=7.4.23

I think its been obsolete since somewhere around 7.02 or 7.00

**Comment by larshp** [comment]
unit test issue reproduced in https://github.com/abapGit/abapGit/pull/6383

**Comment by larshp** [comment]
unit test running in open-abap

**Review by larshp — APPROVED** [review]
looks good, thanks

**Pull request merged**